x-pack/plugins/maps/public/connected_components/mb_map/draw_control/draw_filter_control/draw_filter_control.tsx

```diff
@@ -20,26 +20,26 @@ import {
   roundCoordinates,
 } from '../../../../../common/elasticsearch_util';
 import { getToasts } from '../../../../kibana_services';
-import { DrawControl } from '../draw_control';
+import { DrawControl } from '../';
 import { DrawCircleProperties } from '../draw_circle';
 
 export interface Props {
   addFilters: (filters: Filter[], actionId: string) => Promise<void>;
   disableDrawState: () => void;
   drawState?: DrawState;
-  isDrawingFilter: boolean;
+  filterModeActive: boolean;
   mbMap: MbMap;
   geoFieldNames: string[];
 }
 
 export class DrawFilterControl extends Component<Props, {}> {
-  _onDraw = async (e: { features: Feature[] }) => {
+  _onDraw = () => async (e: { features: Feature[] }) => {
```

Comment:
Why did you wrap _onDraw in another function. This does not seem right.

Reply:
Sorry, I should've thrown a gh comment in here explaining this one. This callback has this shape because of how we use it [here](http://github.com/aaronjcaldwell/kibana/blob/a198ecc215ff81f1bc83271c89fcae3aec08857f/x-pack/plugins/maps/public/connected_components/mb_map/draw_control/draw_feature_control/draw_feature_control.tsx#L39-L39) in the `DrawFeatureControl`. We need access to the `MapboxDraw` instance in order to clean up geometries at the end of the `onDraw` event so it's bound via the first function call. Since the two components share the same callback signature but the `DrawFilterControl` doesn't need access to `MapboxDraw`, it looks strange here.